source/grid/tria_description.cc

```diff
@@ -525,6 +526,380 @@ namespace TriangulationDescription
+          tria.save_user_flags(old_user_flags);
+
+          const_cast<dealii::Triangulation<dim, spacedim> &>(tria)
+            .clear_user_flags();
+
+          // mark all vertices attached to locally owned cells
+          std::vector<bool> vertices_owned_by_locally_owned_cells(
+            tria.n_vertices());
+          for (const auto &cell : tria.active_cell_iterators())
+            if (cell->is_locally_owned() &&
+                static_cast<unsigned int>(
+                  partition[cell->global_active_cell_index()]) == proc)
+              add_vertices_of_cell_to_vertices_owned_by_locally_owned_cells(
+                cell, vertices_owned_by_locally_owned_cells);
+
+          // helper function if a cell is locally relevant (active and
+          // connected to cell via a vertex)
+          const auto is_locally_relevant_on_level = [&](const auto &cell) {
+            for (const auto v : cell->vertex_indices())
+              if (vertices_owned_by_locally_owned_cells[cell->vertex_index(v)])
+                return true;
+            return false;
+          };
+
+          // collect locally relevant cells (including their parents)
+          for (const auto &cell : tria.active_cell_iterators())
+            if (is_locally_relevant_on_level(cell))
+              fill_cell_infos(cell, description_temp_i.cell_infos, partition);
+
+          // collect coarse-grid cells
+          std::vector<bool> vertices_locally_relevant(tria.n_vertices(), false);
+
+          for (const auto &cell : tria.cell_iterators_on_level(0))
+            {
+              if (!cell->user_flag_set())
+                continue;
+
+              // extract cell definition
+              dealii::CellData<dim> cell_data(cell->n_vertices());
+              cell_data.material_id = cell->material_id();
+              cell_data.manifold_id = cell->manifold_id();
+              for (const auto v : cell->vertex_indices())
+                cell_data.vertices[v] = cell->vertex_index(v);
+              description_temp_i.coarse_cells.push_back(cell_data);
+
+              // mark cell vertices as relevant
+              for (const auto v : cell->vertex_indices())
+                vertices_locally_relevant[cell->vertex_index(v)] = true;
+
+              // save translation for corase grid: lid -> gid
+              description_temp_i.coarse_cell_index_to_coarse_cell_id.push_back(
+                cell->id().get_coarse_cell_id());
+            }
+
+          // collect coarse-grid vertices
+          for (unsigned int i = 0; i < vertices_locally_relevant.size(); ++i)
+            if (vertices_locally_relevant[i])
+              description_temp_i.coarse_cell_vertices.emplace_back(
+                i, tria.get_vertices()[i]);
+
+          // restore flags
+          const_cast<dealii::Triangulation<dim, spacedim> &>(tria)
+            .load_user_flags(old_user_flags);
+        }
+
+      // collect description from all processes in a single description
```

Comment:
I don't understand the "all" in this comment. You are referring to the one which hold some of the children/cells (according to some partitioning strategy), not *all* in the MPI universe, right?

Reply:
I have updated the comment!